test/logstream/v2/stream.go

```diff
@@ -74,38 +83,40 @@ func (s *namespaceSource) StartStream(name string, l Callback) (Canceler, error)
 }
 
 func (s *namespaceSource) watchPods() error {
-	wi, err := s.kc.CoreV1().Pods(s.namespace).Watch(s.ctx, metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-
-	go func() {
-		defer wi.Stop()
-		watchedPods := sets.NewString()
+	for _, ns := range s.namespaces {
```

Comment:
So here we have a loop that creates N goroutines each with its own watch per-namespace.
I think it'd simplify a lot if we had namespaceSource watch only single ns... 
And the goroutine can be named.

Reply:
do you mean basically moving the loop into interface.go here 

 func Start(t ti) Canceler {
	once.Do(func() {
		if ns := os.Getenv(system.NamespaceEnvKey); ns != "" {
			...........
			namespaces := strings.Split(ns, ",")
                          for i, namespace := range namespaces {
		          	stream[i] = &shim{logstreamv2.FromNamespaces(context.Background(), kc, namespaces)}
                           }
                         /// TODO some stream merging wrapper  HERE 
		      ...........
	})

	return stream.Start(t)
}

this doesn't seem much simpler , but i am probably missing something in your proposal

Reply:
🤷  — matter of personal opinion, I guess.